python/doc/examples/calibration/bayesian_calibration/plot_imh_python_distribution.py

```diff
@@ -1,38 +1,217 @@
+        if X[0] < self.rangeLow_ or X[0] >= self.rangeUp_:
+            return -ot.SpecFunc.Infinity
+        return self.logPDF_(X)[0]
+
+    def computePDF(self, X):
+        if X[0] < self.rangeLow_ or X[0] >= self.rangeUp_:
+            return 0.0
+        return self.PDF_(X)[0]
+
+    def computeCDF(self, X):
+        if X[0] < self.rangeLow_:
+            return 0.0
+        if X[0] >= self.rangeUp_:
+            return 1.0
+        return ot.GaussLegendre([32]).integrate(self.PDF_, ot.Interval(self.rangeLow_, X[0]))[0]
+
+    def getRealization(self):
+        return self.sampler_.getRealization()
+
+    def getSample(self, n):
+        return self.sampler_.getSample(n)
+
+# %%
+# We create the new distribution that used the Ratio of Uniforms sampler:
+
+
+newDist_RoU = ot.Distribution(NewDistribution_RoU())
+
+# %%
+# We compare the time for sampling the distribution with the Ratio of Uniforms algorithm to
+# the time with the generic sampler. The Ratio of Uniforms algorithm proves to be much more rapid
+# than the generic method.
+sizeRoU = 10000
+sizeGeneric = 100
+
+t0 = monotonic()
+sample_RoU = newDist_RoU.getSample(sizeRoU)
+t1 = monotonic()
+sample_generic = newDist_generic.getSample(sizeGeneric)
+t2 = monotonic()
+
+print('Ratio of Uniforms speed = ', sizeRoU / (t1 - t0), 'rng/s')
+print('Generic algorithm speed = ', sizeGeneric / (t2 - t1), 'rng/s')
+
+# %%
+# Case 2: Direct use the Ratio of Uniforms algorithm
+# --------------------------------------------------
+# In that case, we want to use the :class:`~openturns.experimental.RatioOfUniforms` algorithm to
+# sample :math:`p`. We need to compute the function :math:`\log f` and its range.
+#
+# We create the function :math:`\log f` and the :class:`~openturns.experimental.RatioOfUniforms`:
+log_UnscaledPDF = ot.ComposedFunction(ot.SymbolicFunction('x', 'log(x)'), f)
+ratioAlgo = otexp.RatioOfUniforms(log_UnscaledPDF, rangePDF)
+
+# %%
+# We can draw the exact PDF and the PDF estimated from a sample generated by the Ratio of Uniforms algorithm.
+
+# %%
+# sphinx_gallery_thumbnail_number = 2
+size = 100000
+sample = ratioAlgo.getSample(size)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(0.0)
+ks_algo.setUpperBound(2.0 * pi)
+ks_pdf = ks_algo.build(sample)
+
+g = ks_pdf.drawPDF()
+draw = exact_PDF.draw(0.0, 2.0 * pi, 1001).getDrawable(0)
+draw.setLineWidth(2)
+g.add(draw)
+g.setLegends(['Ratio of Uniforms', 'exact pdf'])
+g.setLegendPosition('topright')
+g.setXTitle('x')
+view = otv.View(g)
+
+# %%
+# By default, the parameter :math:`r=1`. We can get the associated acceptation ratio:
+print('Acceptance ratio = ', ratioAlgo.getAcceptanceRatio())
 
 # %%
-# Independent Metropolis-Hastings
-# -------------------------------
+# We can change the :math:`r` parameter and check the associated acceptance ratio:
+r_new = 0.5
+ratioAlgo.setR(r_new)
+print('New acceptance ratio = ', ratioAlgo.getAcceptanceRatio())
```

Comment:
The value returned by getAcceptanceRatio() seems to be greater than 1:

![image](https://github.com/user-attachments/assets/5efdfbf9-ee73-412a-ae57-0b9da64e4bc2)

Reply:
Fixed, the maximum multistart parameter was not taken into account and only one candidate was used by the optimization algorithm.